S/SymEngine/build_tarballs.jl

```diff
@@ -0,0 +1,58 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder
+
+# Collection of sources required to build SymEngine
+name = "SymEngine"
+version = v"0.5.0"
+sources = [
+    "https://github.com/symengine/symengine/releases/download/v$(version)/symengine-$(version).tar.gz" =>
+    "5d02002f00d16a0928d1056e6ecb8f34fd59f3bfd8ed0009a55700334dbae29b",
+]
+
+# Bash recipe for building across all platforms
+
+script = raw"""
+cd $WORKSPACE/srcdir/symengine-*
+mkdir build
+cd build
+cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
+      -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+      -DBUILD_TESTS=no \
+      -DBUILD_BENCHMARKS=no \
+      -DBUILD_SHARED_LIBS=yes \
+      -DWITH_MPC=yes \
+      -DWITH_SYMENGINE_THREAD_SAFE=yes ..
+make -j${nproc}
+make install
+
+# On Windows, make sure non-versioned filename exists...
+if [[ ${target} == *mingw* ]]; then
+    cp -v ${prefix}/bin/libsymengine-*.dll ${prefix}/bin/libsymengine.dll
+fi
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+a = supported_platforms()
+# For some reason, on MinGW, using a DLL comppiled with a different gcc
+# segfaults. Expand only on Windows. Other OSes work fine.
+deleteat!(a, findfirst(x->x==Windows(:x86_64), a))
+platforms = BinaryBuilder.expand_gcc_versions(Windows(:x86_64))
```

Comment:
I see the program is written in C++, then you probably want to use `expand_cxxstring_abis` on all `supported_platforms()`?

Reply:
There is a C interface which is what's used by the Julia wrapper, `SymEngine.jl`.

Reply:
Anyway `expand_gcc_versions` doesn't exist :slightly_smiling_face:

Reply:
Have you tried to not expand the GCC versions?  How did you test and get segfaults?

Reply:
This script is a copy of https://github.com/symengine/SymEngineBuilder/blob/master/build_tarballs.jl
I tested on appveyor and also tried RDPing to appveyor and dlopening the library on windows x86_64 just crashes the julia interpreter. Eg: https://ci.appveyor.com/project/symengine/symengine-jl/builds/24473789
After expanding gcc versions, it worked.

Reply:
You were using a very old version of BinaryBuilder, I think also the compilers changed quite a lot since then.  I'd suggest to build without expanding and see how it goes, if there are really problems we'll try to fix them later.  Since there is no `expand_gcc_versions` anymore what we're doing here is a bit weird.

Reply:
Sure. Let's try first without it.